train.rb

```diff
@@ -1 +1,61 @@
+class Train
+  attr_reader :number, :type, :count_wagons, :speed
 
+  def initialize(number, type, count_wagons)
+    @number = number
+    @type = type
+    @count_wagons = count_wagons
+    @speed = 0
+  end
+
+  def decrease_speed(value)
+    @speed -= value
+    @speed = 0 if @speed.negative?
+  end
+
+  def increase_speed(value)
+    @speed += value
+  end
+
+  def add_wagon
+    @count_wagons += 1 if @speed.zero?
+  end
+  
+  def remove_wagon
+    @count_wagons -= 1 if @speed.zero? && @count_wagons.positive?
+  end
+
+  def accept_route(route)
+    @route = route
+    @route.stations.first.add_train(self)
+    @current_station = 0
+  end
+
+  def forward_train
+    return nil unless @current_station < @route.size - 1
```

Comment:
- `return` по дефолту возвращает `nil`
- там где удобно рекомендуется предпочитать `if`, а не `unless`
```suggestion
    return if @current_station >= @route.size - 1
```

Reply:
а еще лучше здесь использовать что-то такое
```suggestion
    return unless next_station
```

Reply:
потерялся метод `current_station`
ТЗ: "Возвращать предыдущую станцию, текущую, следующую, на основе маршрута"